pages/algorithms/semantic-role.md

```diff
@@ -0,0 +1,23 @@
+---
+title: Semantic Role 
+key: semantic-role
+---
+
+A semantic association that indicates an object's type. This allows tools to present and support interaction with the object in a manner that is consistent with user expectations about other objects of that type.
+
+[Roles](https://www.w3.org/TR/wai-aria-1.1/#dfn-role) can be [implicit](https://www.w3.org/TR/wai-aria/#implicit_semantics) through the element type or explicit through the [role attribute](https://www.w3.org/TR/html/dom.html#aria-role-attribute). 
+
+The role attribute takes a list of tokens, the semantic role is the first valid role in this list. If none of the tokens are valid, the implicit role will be used instead.
```

Comment:
editorial: replace with "The role attribute takes a list of tokens. The semantic role is the first valid role in this list. If none of the tokens is valid, the implicit role will be used."